spring-integration-core/src/main/java/org/springframework/integration/endpoint/MessageProducerSupport.java

```diff
@@ -222,7 +234,7 @@ protected void subscribeToPublisher(Publisher<? extends Message<?>> publisher) {
 						.map(this::trackMessageIfAny)
 						.doOnComplete(this::stop)
 						.doOnCancel(this::stop)
-						.takeWhile((message) -> isRunning());
+						.takeWhile((message) -> isActive());
```

Comment:
It's not clear why this is different; both flags are set to true in `start()`.

Reply:
Right, but `running` set after `doStart()` call. My idea (and I see in existing components) to have some flag which is `true` before we call `doStart()`. See that `MessageProducerSupport.start()` impl.

Reply:
Right, but why does it matter that it is set after exiting `doStart()`. The requirement is not clear at all.

Reply:
OK. See this `subscribeToPublisher()` details. There is that `subscribe()` word a couple times.
So, essentially we subscribe to the `Flux` in the `doStart()`, but since `isRunning()` returns `false` for us, we stop consuming the `Flux` immediately according that `takeWhile()` contract.

I won't mind any other ideas.

Thanks

Reply:
Ah; OK; got it now - but why not just set `active=true` in `start()` before calling `doStart()`? I think that would be cleaner.

Reply:
Right. That's an idea, too. Although we would need to move `active` flag from the `MessageProducerSupport` down to the `AbstractEndpoint`...
On the other hand I saw similar `active` in the gateway impls, so probably we are on the right track.

Thanks for confirmation! 
Stay tuned...